File: security-oauth2/src/test/java/io/micronaut/security/oauth2/docs/openid/OktaUserDetailsMapper.java
package io.micronaut.security.oauth2.docs.openid;

//tag::clazz[]
import io.micronaut.security.authentication.UserDetails;
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdClaims;
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdTokenResponse;
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdUserDetailsMapper;

import javax.annotation.Nonnull;
import javax.inject.Named;
import javax.inject.Singleton;
import java.util.Collections;

@Singleton
@Named("okta") // <1>
public class OktaUserDetailsMapper implements OpenIdUserDetailsMapper {

    @Override
    @Nonnull
    public UserDetails createUserDetails(String providerName, // <2>
                                         OpenIdTokenResponse tokenResponse, // <3>
                                         OpenIdClaims openIdClaims) { // <4>
        return new UserDetails("name", Collections.emptyList()); // <5>
    }
}
//end::clazz[]

File: security-oauth2/src/test/kotlin/io/micronaut/security/oauth2/docs/openid/OktaUserDetailsMapper.kt
package io.micronaut.security.oauth2.docs.openid

//tag::clazz[]
import io.micronaut.security.authentication.UserDetails
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdClaims
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdTokenResponse
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdUserDetailsMapper

import javax.inject.Named
import javax.inject.Singleton

@Singleton
@Named("okta") // <1>
class OktaUserDetailsMapper : OpenIdUserDetailsMapper {

    override fun createAuthenticationResponse(providerName: String, // <2>
                                              tokenResponse: OpenIdTokenResponse, // <3>
                                              openIdClaims: OpenIdClaims) // <4>
            : UserDetails {
        return UserDetails("name", emptyList()) // <5>
    }
}
//end::clazz[]